### PR TITLE
[CPU] Fix FusionOfTensorOps nullptr

### PR DESCRIPTION
**Pull request by dcaballe**
This PR fixes a compilation crash of an internal models. I'm not familiar with the pass so happy to change the fix or incorporate a test if you can think of an example to exercise this fix.

### CODE REVIEW

**Comment by dcaballe** [comment]
The Nvidia GPU failure doesn't seem to be related?

**Review by MaheshRavishankar — APPROVED** [review]
This looks good. Thanks for the fix.

**Comment by ScottTodd** [comment]
> The Nvidia GPU failure doesn't seem to be related?

Yeah, that's runtime only code so this couldn't have affected it. Weird though, I've seen some of those CTS tests _timeout_, but a _segfault_ is a bit different...

**Comment by dcaballe** [comment]
Let's rerun...

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp [hunk 1]
```diff
@@ -489,8 +489,8 @@ struct FusionOfTensorOpsPass
             if (!reshapeOp)
               return true;
 
-            return isa<linalg::LinalgOp, tensor::UnPackOp,
-                       LinalgExt::UnsetEncodingOp>(
+            return llvm::isa_and_nonnull<linalg::LinalgOp, tensor::UnPackOp,
+                                         LinalgExt::UnsetEncodingOp>(
                 reshapeOp.getSrc().getDefiningOp());
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp:494]: I think it is okay to land it like this. We are probably hitting the case that `reshapeOp.getSrc()` is a bbarg, so getDefiningOp returns nullptr.

**Review by hanhanW — APPROVED** [review]

**Pull request merged**